Source/WebCore/rendering/style/ScrollbarColor.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "StyleColor.h"
+
+namespace WebCore {
+
+struct ScrollbarColor {
+    StyleColor thumbColor { Color(std::nullopt) };
+    StyleColor trackColor { Color(std::nullopt) };
+
+    ScrollbarColor() { };
+    ScrollbarColor(const StyleColor& thumbColor, const StyleColor& trackColor);
+
+    struct MarkableTraits {
+        static bool isEmptyValue(const ScrollbarColor& value)
+        {
+            return !value.thumbColor.absoluteColor().isValid();
+        }
+
+        static ScrollbarColor emptyValue()
+        {
+            return { };
```

Comment:
This needs to be:
```
    return Color { };
```
So we get invalid color instead of `currentColor`.

Reply:
Wont that return a color rather than an empty ScrollbarColor(s) struct? This Markable is for the ScrollbarColor struct right?

Reply:
Would { } not already use the default values assigned inside the struct (both being assigned Color { })?

Reply:
It would

But not that typically we would not want the default value for a `ScrollbarColor` to be the one that will confuse `Markable` instead there would be one value that is the default for general use and another that is the value used for "null" in `Markable`.